src/test/java/com/petros/bibernate/dao/EntityPersisterTest.java

```diff
@@ -313,6 +315,65 @@ public void testDeleteEntityWithNullId(DatabaseType databaseType) {
         assertThrows(BibernateException.class, () -> entityPersister.delete(product), "ID field is null");
     }
 
+    @ParameterizedTest
+    @EnumSource(DatabaseType.class)
+    @DisplayName("Test insert with @ManyToOne relation")
+    void testInsertWithManyToOneRelation(DatabaseType databaseType) {
+        setUpDatabaseType(databaseType);
+
+        Person personInDatabase = new Person();
+        personInDatabase.setId(1L);
+        personInDatabase.setFirstName("NewFirstName");
+
+        Note newNote = new Note();
+        String noteBody = "New note Body!";
+        newNote.setBody(noteBody);
+        newNote.setPerson(personInDatabase);
+
+        Note createdNote = entityPersister.insert(newNote);
+
+        assertNotNull(createdNote);
+        assertNotNull(createdNote.getPerson());
+        assertNotNull(createdNote.getId());
+        assertEquals(noteBody, createdNote.getBody());
+        assertEquals(personInDatabase.getId(), createdNote.getPerson().getId());
+        assertEquals(personInDatabase.getFirstName(), createdNote.getPerson().getFirstName());
```

Comment:
Here you are comparing the same value, which is not related to the value stored in the database. In the database, Oleg will still be stored. To get the real values from the database, you should call the methods:
```
        Note dbNote = entityPersister.findById(Note.class, newNote.getId());
        Person dbPerson = entityPersister.findById(Person.class, 1L);
```
And in order for the changes to be applied to Person, you need to implement cascade in @ManyToOne.

Reply:
Right. Thanks.